src/Mvc/src/Microsoft.AspNetCore.Mvc.NewtonsoftJson/DependencyInjection/MvcJsonMvcCoreBuilderExtensions.cs

```diff
@@ -80,6 +84,30 @@ internal static void AddJsonFormatterServices(IServiceCollection services)
             services.TryAddEnumerable(
                 ServiceDescriptor.Transient<IApiDescriptionProvider, JsonPatchOperationsArrayProvider>());
             services.TryAddSingleton<JsonResultExecutor>();
+
+            var tempDataSerializer = services.FirstOrDefault(f => f.ServiceType == typeof(TempDataSerializer));
```

Comment:
I couldn't get `ServiceCollection.Replace` to work, I'll give it another try if we want to keep this behavior. That said, we could possibly change the default implementation to use `json` instead.